src/common/KokkosKernels_Utils.hpp

```diff
@@ -212,49 +212,26 @@ void get_suggested_vector_size(
 
 }
 
-
-template <typename ExecutionSpace>
-void get_suggested_team_size(
-    int max_allowed_team_size,
-    int suggested_vector_size,
-    int &suggested_team_size_){
-
-    suggested_team_size_ = 1;
-
-#if defined( KOKKOS_ENABLE_SERIAL )
-  if (Kokkos::Impl::is_same< Kokkos::Serial , ExecutionSpace >::value){
-    suggested_team_size_ = 1;
-  }
-#endif
-
-#if defined( KOKKOS_ENABLE_THREADS )
-  if (Kokkos::Impl::is_same< Kokkos::Threads , ExecutionSpace >::value){
-    suggested_team_size_ =  1;
-  }
-#endif
-
-#if defined( KOKKOS_ENABLE_OPENMP )
-  if (Kokkos::Impl::is_same< Kokkos::OpenMP, ExecutionSpace >::value){
-    suggested_team_size_ = 1;
-  }
-#endif
-
-#if defined( KOKKOS_ENABLE_CUDA )
-  if (Kokkos::Impl::is_same<Kokkos::Cuda, ExecutionSpace >::value){
-    suggested_team_size_ = max_allowed_team_size / suggested_vector_size;
-    if(suggested_team_size_ == 0)
-      suggested_team_size_ = 1;
+//Get the best team size for the given functor.
+//If it uses shared memory, the amount used must be available through f.team_shmem_size(n),
+//not through the TeamPolicy. If this is how dynamic shared is set, just use AUTO for the team size.
+template<typename team_policy_t, typename Functor, typename ParallelTag = Kokkos::ParallelForTag>
+int get_suggested_team_size(Functor& f, int vector_size)
```

Comment:
Do we still use vector_size ?

Reply:
Yes, all the kernels that use this first get a preferred vector size (in general, 1 on CPU and 32 on CUDA), and then find a team size so that teamSize * vectorSize = a reasonable CUDA block size. A kernel that doesn't use ThreadVector loops would just pass in 1.